models/order.go

```diff
@@ -235,9 +235,18 @@ func (o *Order) Fill(ctx context.Context, fillSize decimal.Decimal) (isFilled bo
 	return o.IsFilled(), nil
 }
 
-func (o *Order) Kill(ctx context.Context, size decimal.Decimal) error {
+func (o *Order) Lock(ctx context.Context, size decimal.Decimal) error {
```

Comment:
Probably a good idea to write a few tests around these new methods. I remember I messed up my original logic and only caught the bug when I wrote a test